providers/sftp/src/airflow/providers/sftp/sensors/sftp.py

```diff
@@ -97,16 +97,16 @@ def poke(self, context: Context) -> PokeReturnValue | bool:
         else:
             actual_files_to_check = [self.path]
 
-        for actual_file_to_check in actual_files_to_check:
-            try:
-                mod_time = self.hook.get_mod_time(actual_file_to_check)
-                self.log.info("Found File %s last modified: %s", actual_file_to_check, mod_time)
-            except OSError as e:
-                if e.errno != SFTP_NO_SUCH_FILE:
-                    raise AirflowException from e
-                continue
-
-            if self.newer_than:
+        if self.newer_than:
```

Comment:
Can you please add test that tests it? Simply check that no get_mod_time is called when newer_than is None/False.

Reply:
added